src/main/nav_create_simulated_image.py

```
@@ -1682,45 +2245,59 @@ def _update_display(self) -> None:
+        rings = self.sim_params.get('rings', [])
+        if ring_masks and rings:
+            for j, ring_mask in enumerate(ring_masks):
+                if j < len(rings):
+                    range_val = float(rings[j].get('range', 1000.0 + j))
+                    objects.append((range_val, 'ring', j, ring_mask))
+
+        # Sort by range (near to far = ascending range)
+        objects.sort(key=lambda x: x[0])
+
+        # Check objects in order (near to far), select first match
+        for range_val, kind, idx, mask in objects:
```

Ruff (0.14.11)

Loop control variable `range_val` not used within loop body

(B007)